ruff_cgx/linter.py

```diff
@@ -47,41 +47,71 @@ def _prepare_content_for_linting(
     return virtual_content, parsed, script_content
 
 
-def lint_file(path, fix=False, **_):
+def lint_file(path, fix=False):
     """
-    Lint a CGX file using ruff (CLI version).
+    Lint a CGX file and return structured data.
 
     Args:
         path: Path to the CGX file
         fix: Whether to fix issues (default: False)
 
     Returns:
-        Exit code (0 for success, non-zero for errors)
+        dict with:
+        - path: Path object
+        - was_fixed: bool (True if fixes were applied)
+        - diagnostics: List[Diagnostic] (remaining issues)
+        - diagnostics_before_fix: List[Diagnostic] (issues before fix, if fix=True)
+        - success: bool (True if no errors)
     """
     path = Path(path)
     content = path.read_text(encoding="utf-8")
 
     # Prepare content for linting
     virtual_content = _prepare_content_for_linting(content)
     if virtual_content is None:
-        return 1
+        return {
+            "path": path,
+            "was_fixed": False,
+            "diagnostics": [],
+            "diagnostics_before_fix": [],
+            "success": False,
+        }
 
     virtual_content, parsed, script_content = virtual_content
 
-    # Run ruff check with full output (for CLI)
-    result, temp_path, fixed_content = run_ruff_check(
-        virtual_content, output_format="full", fix=fix
-    )
-
-    # Replace temp file path with actual path in output
-    stdout = result.stdout.replace(str(temp_path), str(path)).strip()
-    print(stdout)  # noqa: T201
-
-    # If fix was requested and we got fixed content, apply it back to the file
-    if fix and fixed_content:
+    # Parse initial diagnostics
+    diagnostics_before_fix = lint_cgx_content(content)
+
+    # If not fixing, just return the diagnostics
+    if not fix:
+        return {
+            "path": path,
+            "was_fixed": False,
+            "diagnostics": diagnostics_before_fix,
+            "diagnostics_before_fix": diagnostics_before_fix,
+            "success": len(diagnostics_before_fix) == 0,
+        }
+
+    # Run ruff check with fix
+    _, fixed_content = run_ruff_check(virtual_content, fix=fix)
+
+    # Apply fixes if we got fixed content
+    was_fixed = False
+    if fixed_content:
         _apply_fixes_to_file(path, content, parsed, script_content, fixed_content)
+        was_fixed = True
 
-    return result.returncode
+    # Re-lint to get remaining diagnostics
+    fixed_file_content = path.read_text(encoding="utf-8")
+    diagnostics = lint_cgx_content(fixed_file_content)
```

Comment:
We can probably prevent the extra lint here

Reply:
If the output should act like ruff's own output, then we'll need to run ruff 3 times... one time for figuring out how many fixable problems there are, then to apply the actual fix, and then a last time to get the remaining fixes (can't just filter the results from the first run, since the line numbers and such might have changed because of the fixes).